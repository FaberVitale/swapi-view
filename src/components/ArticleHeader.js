//@flow
import * as React from "react";
import theme from "../styles/theme";

type Props = {
  title: string,
  navigation?: React.Element<*>
};

/* Header that renders on the bottom-left the title of the article
 * and on the top-right a navigation widget passed as prop
 * 
 * we use PureComponent to prevent a re-render if title is the same 
 * and navigation is undefined 
*/
class ArticleHeader extends React.PureComponent<Props> {
  render() {
    return (
      <header className="article-header justify-space-between flex-column align-items-center">
        <div
          aria-hidden={!this.props.navigation}
          className="article-header-group flex-row justify-flex-end header-nav"
        >
          {this.props.navigation}
        </div>
        <div className="article-header-group justify-flex-start">
          <h2 className="article-header-title" style={theme.h2}>
            {this.props.title}
          </h2>
        </div>
      </header>
    );
  }
}

export default ArticleHeader;
