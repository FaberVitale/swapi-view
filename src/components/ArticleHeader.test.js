import ArticleHeader from "./ArticleHeader";
import renderer from "react-test-renderer";
import React from "react";

describe("components/ArticleHeader", () => {
  const tree = renderer
    .create(
      <ArticleHeader title="test-test" navigation={<span>{"test"}</span>} />
    )
    .toJSON();
  it(" matches the snapshot", () => {
    expect(tree).toMatchSnapshot();
  });
});
