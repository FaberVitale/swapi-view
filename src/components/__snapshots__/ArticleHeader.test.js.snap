// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ArticleHeader  matches the snapshot 1`] = `
<header
  className="article-header justify-space-between flex-column align-items-center"
>
  <div
    aria-hidden={false}
    className="article-header-group flex-row justify-flex-end header-nav"
  >
    <span>
      test
    </span>
  </div>
  <div
    className="article-header-group justify-flex-start"
  >
    <h2
      className="article-header-title"
      style={
        Object {
          "color": "#ffeb3b",
        }
      }
    >
      test-test
    </h2>
  </div>
</header>
`;
